latch/registry/table.py

```diff
@@ -0,0 +1,290 @@
+                    catalogExperiment(id: $argTableId) {
+                        catalogSamplesByExperimentId(
+                            condition: { removed: false }
+                            first: $argPageSize
+                            after: $argAfter
+                        ) {
+                            nodes {
+                                id
+                                name
+                                catalogSampleColumnDataBySampleId {
+                                    nodes {
+                                        data
+                                        key
+                                    }
+                                }
+                            }
+                            pageInfo {
+                                endCursor
+                                hasNextPage
+                            }
+                        }
+                    }
+                }
+                """,
+                {
+                    "argTableId": self.id,
+                    "argAfter": end_cursor,
+                    "argPageSize": page_size,
+                },
+            )["catalogExperiment"]["catalogSamplesByExperimentId"]
+
+            records = data["nodes"]
+            end_cursor = data["pageInfo"]["endCursor"]
+            has_next_page = data["pageInfo"]["hasNextPage"]
+
+            output: List[Record] = []
+            errors: List[Exception] = []
+
+            for record in records:
+                record_data = {
+                    node["key"]: node["data"]
+                    for node in record["catalogSampleColumnDataBySampleId"]["nodes"]
+                }
+
+                values: Dict[str, object] = {}
+                valid = True
+
+                try:
+                    for column in self.get_columns(load_if_missing=True):
+                        key = column["key"]
+                        typ = column["type"]
+
+                        data_point = record_data.get(key)
+                        if data_point is not None:
+                            values[key] = to_python_literal(
+                                data_point,
+                                typ["type"],
+                            )
+
+                            if isinstance(values[key], RegistryInvalidValue):
+                                valid = False
+
+                        if key not in values:
+                            if not typ["allowEmpty"]:
+                                valid = False
+                            values[key] = registry_empty_cell
+                except RegistryTransformerException as e:
+                    # todo(ayush): raise immediately and prompt for confirmation
+                    # if the user wants the rest of the page to be processed?
+                    errors.append(
+                        ValueError(
+                            f"record {record['id']} ({record['name']}) is invalid: {e}"
```

Comment:
```suggestion
                            f"record {record['id']} ({record['name']}) is invalid"
```
and set `__cause__`, `__context__` to `e`